int16 code
int16 SUCCESS = 0
int16 IK_FAILURE = -1
int16 INPUT_FAILURE = -2
int16 TF_FAILURE = -3
